sequencing/ATACSeq/jobSubmission/1_batchRunPreAnalysis.sh

```diff
@@ -0,0 +1,159 @@
+## LAST UPDATE: February 2024                                                                                         ||
+## ===================================================================================================================##
+## EXECUTION: sbatch --array= ./sequencing/ATACSeq/jobSubmission/1_batchRunPreAnalysis.sh <projectName> <option>      ||
+## - execute from scripts directory                                                                                   ||
+##                                                                                                                    ||
+## INPUTS:                                                                                                            || 
+## --array -> Number of jobs to run. Will select sample(s) corresponding to the number(s) input                       ||
+## $1 -> <projectName> Name of project                                                                                ||
+## $2 -> <option> Specify step to run: FASTQC, TRIM, ALIGN or ENCODE. Can be combined. Default is to                  ||
+## run all                                                                                                            ||
+##                                                                                                                    ||
+## DESCRIPTION: This script performs the first step in the ATAC-seq pipeline: pre-analysis. This                      ||
+## includes quality control of raw reads (FASTQC), trimming the reads (TRIM), aligning reads to                       ||
+## reference genome (ALIGN) and calculate ENCODEQC metrics (ENCODE).                                                  ||
+##                                                                                                                    ||
+## REQUIRES:                                                                                                          ||
+## - File in ${METADIR}/samples.txt that lists sample names.                                                          ||
+## ===================================================================================================================##
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
+
+## print start date and time when script runs
+echo Job started on:
+date -u
+
+## needs to be executed from the scripts folder
+echo "Changing Folder to: " $SLURM_SUBMIT_DIR
+cd $SLURM_SUBMIT_DIR || exit 1
+
+## load config file provided on command line related to the specified project
+echo "Loading config file for project: " $1
+export PROJECT=$1
+source ./ATACSeq/config/config.txt 
+
+## check directories
+echo "Project directory is: " $DATADIR 
+echo 'Script is running from directory: ' ${SCRIPTDIR}
+
+##check array specified and exit if not
+if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
+then 
+    { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
+fi
```

Comment:
Instead of checking if the array is submitted as an array. We could add a SLURM directive at the top of the script:
`#SBATCH --array=1-4` (or similar)

Reply:
Nevermind, I've read further and it seems that the user should specify this themselves as otherwise not all files will be processed.
I guess in the future this could be improved by introducing loops and splitting the samples up into each of the jobs in the array.